loader/src/utils.js

```diff
@@ -523,4 +523,117 @@ module.exports = {
 
     return undefined;
   },
+
+  /**
+   * Group locations by external ID. Locations will appear in more than one group
+   * if more than one external ID is shared.
+   * @param {Array} locations
+   * @param {Array<string>} [systems] Only consider these external ID systems.
+   * @param {boolean} [unpadIds] Compare unpadded versions of numeric IDs.
+   * @returns {Map<string,Set>}
+   */
+  groupByExternalId(locations, systems = null, unpadIds = false) {
+    const byExternalId = new Map();
+    let counter = 0;
+
+    // Make a locations lookup and an external IDs lookup
+    for (const location of locations) {
+      let inAGroup = false;
+      for (const [system, value] of location.external_ids) {
+        // Skip not-quite-unique systems
+        if (system === "vtrcks") continue;
+        if (system === "npi_usa") continue;
+        // Skip internal identifiers
+        if (system === "univaf_v0") continue;
+        if (system === "univaf_v1") continue;
+
+        // Only for specified systems
+        if (systems && !systems.includes(system)) continue;
+
+        const mergeValue = unpadIds ? value.replace(/^0+(\d+)$/, "$1") : value;
+        const simpleId = `${system}:${mergeValue}`;
+        let locationSet = byExternalId.get(simpleId);
+        if (!locationSet) {
+          locationSet = new Set();
+          byExternalId.set(simpleId, locationSet);
+        }
+        locationSet.add(location);
+        inAGroup = true;
+      }
+
+      // If a location didn't have any of the ID systems used for grouping, it
+      // still needs to get included in the output.
+      if (!inAGroup) {
+        byExternalId.set(`__UNIQUE__:${counter}`, new Set([location]));
+        counter++;
+      }
```

Comment:
This (and a corresponding removal of one line from `mergeGroupsWithCommonLocations`) is the only change I made from the server script. In the server, we only want to find the duplicates and mess with them, but here we don’t want to lose track of the non-duplicates (we’re just trying to remove the duplicates from a list).